tests/test_subprocesses.py

```diff
@@ -99,3 +99,36 @@ async def test_process_new_session_sid() -> None:
 
     result = await run_process(cmd, start_new_session=True)
     assert result.stdout.decode().strip() != str(sid)
+
+
+async def test_run_process_connect_to_file(tmp_path: Path) -> None:
+    stdinfile = tmp_path / "stdin"
+    stdinfile.write_text("Hello, process!\n")
+    stdoutfile = tmp_path / "stdout"
+    stderrfile = tmp_path / "stderr"
+    with stdinfile.open("rb") as fin, \
+            stdoutfile.open("wb") as fout, \
+            stderrfile.open("wb") as ferr:
+        async with await open_process(
+            [
+                sys.executable, '-c',
+                'import sys; txt = sys.stdin.read().strip(); '
+                'print("stdin says", repr(txt), "but stderr says NO!", file=sys.stderr); '
+                'print("stdin says", repr(txt), "and stdout says YES!")'
+            ],
+            stdin=fin,
+            stdout=fout,
+            stderr=ferr,
+        ) as p:
+            assert await p.wait() == 0
+    assert stdoutfile.read_text() == "stdin says 'Hello, process!' and stdout says YES!\n"
```

Comment:
Add a blank line after a control block if the code continues on the same level.
```suggestion

    assert stdoutfile.read_text() == "stdin says 'Hello, process!' and stdout says YES!\n"
```